core/sql/sqlcomp/CmpSeabaseDDLcommon.cpp

```diff
@@ -6019,6 +6181,93 @@ short CmpSeabaseDDL::updateTextTable(ExeCliInterface *cliInterface,
   return 0;
 }
 
+short CmpSeabaseDDL::updateTextTableWithBinaryData
+(ExeCliInterface *cliInterface,
+ Int64 objUID, 
+ ComTextType textType, 
+ Lng32 subID, 
+ char * inputData,
+ Int32 inputDataLen,
+ NABoolean withDelete)
+{
+  Lng32 cliRC = 0;
+  if (withDelete)
+    {
+      // Note: It might be tempting to try an upsert instead of a
+      // delete followed by an insert, but this won't work. It is
+      // possible that the metadata text could shrink and take fewer
+      // rows in its new form than the old. So we do the simple thing
+      // to avoid such complications.
+      cliRC = deleteFromTextTable(cliInterface, objUID, textType, subID);
+      if (cliRC < 0)
+        {
+          return -1;
+        }
+    }
+
+  // convert input data to utf8 first.
+  ComDiagsArea * diagsArea = CmpCommon::diags();
+  char * inputDataUTF8 = new(STMTHEAP) char[inputDataLen*4];
+  Lng32 inputDataLenUTF8 = 0;
+  ex_expr::exp_return_type rc =
+    convDoIt(inputData,
+             inputDataLen,
+             REC_BYTE_F_ASCII,
+             0,
+             (Int32)CharInfo::ISO88591,
+             inputDataUTF8,
+             inputDataLen*4,
+             REC_BYTE_V_ASCII,
+             inputDataLen,
+             (Int32)CharInfo::UTF8,
+             (char*)&inputDataLenUTF8,
+             sizeof(Lng32),
+             STMTHEAP,
+             &diagsArea,
+             CONV_ASCII_F_V);
+  if ((rc != ex_expr::EXPR_OK) ||
+      (inputDataLenUTF8 <= 0))
+    {
+      return -1;
+    }
+  
+  Int32 maxLen = TEXTLEN;
+  char queryBuf[1000];
+  Lng32 numRows = (inputDataLenUTF8 / maxLen) + 1;
+  Lng32 currPos = 0;
+  Int32 currDataLen = 0;
+
+  for (Lng32 i = 0; i < numRows; i++)
+    {
+      NAString temp;
+
+      if (i < numRows-1)
+        currDataLen = maxLen;
+      else
+        currDataLen = inputDataLenUTF8 - currPos;
+
+      str_sprintf(queryBuf, "insert into %s.\"%s\".%s values (%Ld, %d, %d, %d, 0, cast(? as char(%d bytes) character set utf8 not null))",
+                  getSystemCatalog(), SEABASE_MD_SCHEMA, SEABASE_TEXT,
+                  objUID,
+                  textType,
+                  subID,
+                  i,
+                  currDataLen);
+      cliRC = cliInterface->executeImmediateCEFC
+        (queryBuf, &inputData[currPos], currDataLen, NULL, NULL, NULL);
```

Comment:
This could chop a UTF-8 character in half. If that would happen, we would get an error. See method lightValidateUTF8Str() in file sql/common/csconvert.h for finding a valid spot to cut.